src/main/java/com/serenitydojo/Cart.java

```diff
@@ -0,0 +1,51 @@
+package com.serenitydojo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cart {
+
+    private final Catalog catalog;
+    private CartItem item;
+    private final List<CartItem> items = new ArrayList<>();
+
+    public Cart(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    public int getNumberofItems() {
+        return items.size();
+    }
+
+    public void addtoCart(CartItem item) {
+        items.add(item);
+
+    }
+
+    public double getTotalUnits() {
+        return items.stream()
+                .mapToDouble(item -> item.getQuantity())
+                .sum();
+
+    }
+
+
+    public double getRunningTotal() {
+        return items.stream()
+                .mapToDouble(item -> priceOfFruit(item))
+                .sum();
+
+    }
+
+
+    private double priceOfFruit(CartItem item) {
```

Comment:
> - When you buy 5 kilos or more of any fruit, you get a 10% discount.

i read that a bit differently, like "if any fruit > 5 quantity, then runningTotal * 0.9
but your reading of that requirement is fine too :+1: